lib/yard/cli/yardopts_command.rb

```diff
@@ -65,11 +65,14 @@ def yardopts_options(opts)
 
       private
 
-      # Parses the .yardopts file for default yard options
+      # Parses the .yardopts file for default yard options,
+      # ignoring comment lines beginning with // or #
       # @return [Array<String>] an array of options parsed from .yardopts
       def yardopts(file = options_file)
         return [] unless use_yardopts_file
-        File.read_binary(file).shell_split
+        contents = File.read_binary(file)
+        opts = contents.gsub(%r{(#|//).*\n?}, "")
```

Comment:
We're going to need a requirement that comments start at the beginning of the line (incl. optional whitespace), otherwise a valid option like the following will not work:

```
--title "# a title!"
```

Or even:

```
lib/*/*.rb
lib//*.rb
```

Unfortunately this would be a hard requirement.

I would recommend tests for these edge cases too.